pkg/agent/controller/agent.go

```diff
@@ -210,6 +259,33 @@ func (a *Controller) Start(stopCh <-chan struct{}) error {
 	return nil
 }
 
+func (a *Controller) remoteServiceExportLister(transform func(si *mcsv1a1.ServiceExport) runtime.Object) []runtime.Object {
```

Comment:
confirming my understanding - this lists all ServiceExports on broker, not those that are for a specific service, correct?
If we need only a subset (e.g., those owned by the local cluster) shouldn't the Lister include a filter (e.g., broker namespace is already enforced by the Syncer framework, but can add LabelSelectors if needed)

Reply:
Yes, its lists all exports and then we filter the list in line 244.
The reason I didn't use a label selector is because they usually do resource listing not directly using k8s API but using some cache in resource_syncer - see line 263. I admit I still don't fully understand this cache and why its required so I preferred to be cautious, keep using it and avoid such a change at this stage.